poc_mobile_verifier/src/ingest.rs

```diff
@@ -0,0 +1,74 @@
+use crate::Result;
+use chrono::{DateTime, Utc};
+use file_store::{
+    heartbeat::{CellHeartbeat, CellHeartbeatIngestReport},
+    speedtest::{CellSpeedtest, CellSpeedtestIngestReport},
+    traits::MsgDecode,
+    FileStore, FileType,
+};
+use futures::{stream, Stream, StreamExt};
+use std::ops::Range;
+
+pub async fn ingest_heartbeats(
+    file_store: &FileStore,
+    epoch: &Range<DateTime<Utc>>,
+) -> Result<impl Stream<Item = CellHeartbeat>> {
+    Ok(file_store
+        .source(
+            stream::iter(
+                file_store
+                    .list_all(FileType::CellHeartbeatIngestReport, epoch.start, epoch.end)
+                    .await?,
+            )
+            .map(Ok)
+            .boxed(),
+        )
+        .filter_map(|msg| async move {
+            msg.map_err(|err| {
+                tracing::error!("Error fetching heartbeat ingest report: {:?}", err);
+                err
+            })
+            .ok()
+        })
+        .filter_map(|msg| async move {
+            CellHeartbeatIngestReport::decode(msg).map_or_else(
+                |err| {
```

Comment:
i didn't think the default closure in `map_or_else` took any arguments?

Reply:
It doesn't if the type is an option.